src/vanilla.ts

```diff
@@ -10,10 +22,35 @@ export type PartialState<
 > =
   | (Pick<T, K1> | Pick<T, K2> | Pick<T, K3> | Pick<T, K4> | T)
   | ((state: T) => Pick<T, K1> | Pick<T, K2> | Pick<T, K3> | Pick<T, K4> | T)
+
+/**
+ * @deprecated Use `(t: T) => U` instead of `StateSelector<T, U>`.
+ * `StateSelector` will be removed in next major.
+ */
 export type StateSelector<T extends State, U> = (state: T) => U
+
+/**
+ * @deprecated Use `(a: T, b: T) => boolean` instead of `EqualityChecker<T>.
+ * `EqualityChecker` will be removed in next major.
+ */
 export type EqualityChecker<T> = (state: T, newState: T) => boolean
+
+/**
+ * @deprecated Use `(state: T, prevState: T) => void` instead of `StateListener<T>`.
+ * `StateListener` will be removed in next major.
+ */
 export type StateListener<T> = (state: T, previousState: T) => void
+
+/**
+ * @deprecated Use `(slice: T, prevSlice: T) => void` instead of `StateSliceListener<T>`.
+ * `StateSliceListener` will be removed in next major.
+ */
 export type StateSliceListener<T> = (slice: T, previousSlice: T) => void
+
+/**
+ * @deprecated Use `Store<T>['subscribe']` instead of `Subscribe<T>`.
```

Comment:
I don't dislike this, but people would get confused. Better to stay. Same for `SetState` and `GetState`.